src/main/java/io/lakefs/PathMapper.java

```diff
@@ -0,0 +1,307 @@
+                .filter(pm -> pm.isAppropriateMapping(path)).findFirst();
+        return appropriateMap.isPresent() ? appropriateMap.get() : null;
+    }
+
+    /**
+     * PathMapping represents a pair of mapping configurations, that define source and destination prefixes to
+     * be replaced.
+     */
+    private static class PathMapping {
+
+        private MappingConfig srcPrefix;
+        private MappingConfig dstPrefix;
+        private int priority;
+        private String fromScheme;
+        private boolean defaultMapping;
+
+        public PathMapping(MappingConfig srcPrefix, MappingConfig dstPrefix) {
+            this(srcPrefix, dstPrefix, false);
+        }
+
+        public PathMapping(MappingConfig srcPrefix, MappingConfig dstPrefix, boolean defaultMapping) {
+            this.defaultMapping = defaultMapping;
+            this.srcPrefix = srcPrefix;
+            this.dstPrefix = dstPrefix;
+            if (!srcPrefix.getFromScheme().equals(dstPrefix.getFromScheme())) {
+                LOG.error("src and dst schemes must match, cannot create PathMapping. src:"
+                        + srcPrefix.getFromScheme() + " dst: " + dstPrefix.getFromScheme());
+            }
+            this.fromScheme = srcPrefix.getFromScheme();
+            if (!this.defaultMapping) {
+                if (srcPrefix.getPriority() != dstPrefix.getPriority()) {
+                    LOG.error("src and dst indices must match, cannot create PathMapping. src:"
+                            + srcPrefix.getPriority() + " dst: " + dstPrefix.getPriority());
+                }
+                this.priority = srcPrefix.getPriority();
+            }
+        }
+
+        /**
+         * Checks if this path mapping is appropriate for a {@link Path}.
+         * A path mapping considered appropriate for a path if the path starts with the path mapping's source prefix.
+         *
+         * @param p the path to check whether a mapping is appropriate for
+         * @return true if an appropriate, false otherwise
+         */
+        public boolean isAppropriateMapping(Path p) {
+            String str = p.toString();
+            return str.startsWith(srcPrefix.getValue());
+        }
+
+        public MappingConfig getSrcPrefix() {
+            return srcPrefix;
+        }
+
+        public MappingConfig getDstPrefix() {
+            return dstPrefix;
+        }
+
+        public String getFromScheme() {
+            return fromScheme;
+        }
+
+        public int getPriority() {
+            return priority;
+        }
+
+        @Override
+        public String toString() {
+            return "srcPrefix=" + this.srcPrefix + " dstPrefix=" + this.dstPrefix;
+        }
+    }
+
+    /**
+     * A class that represents a parsed Path mapping configurations supported by RouterFileSystem.
+     * Mapping configuration form is: routerfs.mapping.${fromFsScheme}.${mappingIdx}.${replace/with}=value.
```

Comment:
```suggestion
     * Mapping configuration form is: routerfs.mapping.${fromFsScheme}.${mappingIdx}.${replace/with}=prefix.
```
In other places as well

Reply:
done